### ISSUE TITLE
Investigate `_series_id` model

### ISSUE THREAD

**Issue by pauldix** [influxdata/influxdb#24815]
I'm calling this the _series_id model of mapping LP to the 3.0 table model. The idea goes like this:

- Every row has a `_series_id` which is a 32 byte SHA265 hash of the tagset (tags appear in lexicographical order)
- Data is stored into Parquet files, deduped and sorted by `_series_id`
- Tags columns just like fields. We keep the distinction in the metadata for compatibility with InfluxQL. They need not be dictionary encoded
- String columns (which include tags) will only use dictionary encoding when being persisted if their cardinality is < N (need to figure out what a good value of N is)

My thinking is that this will make sorting easier and it'll keep it consistent across the table for all time, even when new columns are introduced. This will also give an entry point later to create indexes on other columns for value -> list of series ids. This could help with single or low cardinality series lookups.

This will also remove the mandatory dictionary encoding for all tag columns, which can be problematic with high cardinality tags.

Lastly, this is something that will be used in the 3.0 data model that I'm envisioning. So this will create a bridge between the 1.x/2.x data model and 3.x.

Here's a list of tasks I think we'll need to investigate this model:
```[tasklist]
### Tasks
- [ ] https://github.com/influxdata/influxdb/issues/24821
- [ ] https://github.com/influxdata/influxdb/issues/24845
- [ ] https://github.com/influxdata/influxdb/issues/24919
- [ ] https://github.com/influxdata/influxdb/issues/24909
- [ ] https://github.com/influxdata/influxdb/issues/24822
- [x] Investigate how to have just series_id as a key when working with `iox_query` package
- [ ] Investigate how to have tags that are not dictionary encoded and not part of primary key (just another column? How to work with InfluxQL?)
```

**Comment by hiltontj** [comment]
A fair bit of work has been done on the open sub-issues listed above. Much of the work is sitting on issues and branches/PRs that have not been merged, so I wanted to summarize that here.

### #24845 - Support column type for storing bytes

* Extended the `schema` definitions to support a `SeriesId` column type for storing a `FixedSizeBinary`, which is treated as a primary key. Changes can be seen in `influxdb3_core` on https://github.com/influxdata/influxdb3_core/pull/12
* The `_series_id` is stored in parquet using `DELTA_BYTE_ARRAY` encoding
* Measure the impact on parquet file size of adding the `_series_id` column https://github.com/influxdata/influxdb/issues/24845#issuecomment-2051982622. Outcomes:
  * the impact on size varies with cardinality, so compaction would reduce its impact
  * takes about as much space as a float field

### #24919 - Benchmark sort/dedupe by tags vs. `_series_id`

* `_series_id` improves compaction time as the number of tags in the data set increases. This trend is consistent in response to:
  * varying thread count: https://github.com/influxdata/influxdb/issues/24919#issuecomment-2066605590
  * varying cardinality: https://github.com/influxdata/influxdb/issues/24919#issuecomment-2066902271
  * varying tag length: https://github.com/influxdata/influxdb/issues/24919#issuecomment-2070960902
* `_series_id` can improve maximum memory usage during compaction when there are a larger number of tags in the data set, but can degrade maximum memory usage when there are few tags: https://github.com/influxdata/influxdb/issues/24919#issuecomment-2070960902

**Comment by pauldix** [comment]
I'm closing this out for now. I think that what the testing here showed overall was that there was some storage impact, which would be greatly mitigated by reshuffling the data so that a single series is clustered together in fewer Parquet files. Compaction was improved in cases where there were more than 2 tags using less CPU and RAM.

Originally, I thought we'd have to use this to build an index to do quick single series lookups, but I've found a better pathway to get to that. I also speculated that the gains for compaction would be greater than these tests have shown.

One of the other concerns I had about being able to consistently sort data over time using the series_id likely won't be a problem in the v3 data model.

Basically, the gains didn't seem impactful enough to continue down this path. We can revisit in the future if it seems like something that will be useful.
